packages/boot/test/bootstrapTests/test-addAssets.ts

```diff
@@ -0,0 +1,196 @@
+import { test as anyTest } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+
+import type { TestFn } from 'ava';
+
+import { TimeMath } from '@agoric/time';
+import {
+  LiquidationTestContext,
+  makeLiquidationTestContext,
+} from './liquidation.ts';
+
+const test = anyTest as TestFn<LiquidationTestContext>;
+
+const auctioneerPath = 'published.auction';
+
+test.before(async t => {
+  t.context = await makeLiquidationTestContext(t);
+  const proposal = await t.context.buildProposal({
+    package: 'builders',
+    packageScriptName: 'build:add-STARS-proposal',
+  });
+
+  t.log('installing proposal');
+  // share a single proposal so tests don't stomp on each other's files; It has
+  // to be edited by each so as not to re-use keywords.
+  for await (const bundle of proposal.bundles) {
+    await t.context.controller.validateAndInstallBundle(bundle);
+  }
+  t.log('installed', proposal.bundles.length, 'bundles');
+  // @ts-expect-error override
+  t.context.proposal = proposal;
+  t.log('installed', proposal.bundles.length, 'bundles');
+});
+
+test.after.always(t => {
+  // This will fail if a subset of tests are run. It detects that three
+  // collaterals were added to the auction after ATOM.
+  t.like(t.context.readLatest(`${auctioneerPath}.book3`), {
+    currentPriceLevel: null,
+  });
+  return t.context.shutdown && t.context.shutdown();
+});
+
+test('addAsset to quiescent auction', async t => {
+  const {
+    advanceTimeTo,
+    readLatest,
+    // @ts-expect-error override
+    proposal,
+  } = t.context;
+
+  // stringify, modify and parse because modifying a deep copy was fragile.
```

Comment:
please put this logic in to the context so in the test it can be,
```
  const proposal = t.context.getCollateralProposal('COMETS');
```

Reply:
done